service/controller/v5/resource/chart/resource.go

```diff
@@ -84,6 +86,11 @@ func isChartModified(a, b ChartState) bool {
 		return true
 	}
 
+	// ReleaseStatus do not match so we need to update the Helm Release.
+	if a.ReleaseStatus != b.ReleaseStatus {
```

Comment:
We should also check what are the implications when installating the helm release and status == `PENDING`.

Reply:
good point, we have to handle state transition. so let's move this logic outside this function.